src/Bridge/Symfony/Bundle/DependencyInjection/ApiPlatformExtension.php

```diff
@@ -218,7 +218,9 @@ private function registerMetadataConfiguration(ContainerBuilder $container, arra
 
         $container->getDefinition('api_platform.metadata.extractor.xml')->addArgument($xmlResources);
 
-        if (class_exists(Annotation::class)) {
+        // The "annotation_reader" service is only available if DoctrineBundle is loaded.
+        $bundles = $container->getParameter('kernel.bundles');
+        if (class_exists(Annotation::class) && isset($bundles['DoctrineBundle'])) {
```

Comment:
Is this check on bundles really mandatory?

Reply:
I know it's a bit weird but in theory it's possible to require "doctrine/annotation" without "doctrine/doctrine-bundle", in that case the class exists but not the service definition.

Reply:
At first it does look cleaner but it's not as reliable. Again, in theory, you can install a bundle with composer but not register it in Symfony's kernel. But since it's an edge case, maybe it's acceptable to simply check for class existance?

Reply:
I'd need some expert hands to answer this as I'm not qualified enough :p

Reply:
I agree with @fgrandjean it should not be tested alone, since it could lead to unexpected behaviours (as the bundle may not be in the Kernel). So I guess it's better to test that the bundle is loaded.

Reply:
I'm for keeping the current code too.